elastic/kibana
# [Guided onboarding] Fix API requests spam when request fails

## Summary
Fixes https://github.com/elastic/kibana/issues/145434

This PR fixes the API spam when a request fails. The API endpoint throws an error when the user doesn't have the correct permissions for the saved objects, which is the intended logic. We will handle the correct permissions for the plugin's saved objects in 8.7+ release. This PR changes how we handle a failed API request so that the Observable is initialised correctly with the new plugin state. So there are no repeated API requests as before.

### Screenshots

API requests spam before

![image](https://user-images.githubusercontent.com/6935300/200026086-305ce3d6-776c-46e9-9b76-033ba4e87cf1.png) 

## How to test
1. Replace line 26 in the file `/src/plugins/guided_onboarding/server/routes/plugin_state_routes.ts` ([link](https://github.com/elastic/kibana/blob/913527036e107645d024c251ceb9902252635e9f/src/plugins/guided_onboarding/server/routes/plugin_state_routes.ts#L26)) with this line `return response.badRequest();` to simulate a failed request.
2. Start Kibana with `xpack.cloud.id: test` in the config.yml file.
3. Navigate to /app/security/alerts and make sure there is only 1 failed request to /api/guided_onboarding/state.
